Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php

```diff
@@ -0,0 +1,521 @@
+ */
+
+namespace Symfony\CS\Tests\Fixer\Contrib;
+
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class Php4ConstructorFixerTest extends AbstractFixerTestBase
+{
+    public function testNamespaces()
+    {
+        $input = <<<'EOF'
+<?php
+
+namespace Baz;
+
+class Foo
+{
+    public function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    public function Foo($bar)
+    {
+        var_dump(2);
+    }
+}
+EOF;
+
+        $this->makeTest($input);
+    }
+
+    public function testNamespaces2()
+    {
+        $input = <<<'EOF'
+<?php
+
+namespace Baz
+{
+    class Foo
+    {
+        public function Foo($bar)
+        {
+            var_dump(1);
+        }
+
+        public function Foo($bar)
+        {
+            var_dump(2);
+        }
+    }
+}
+
+class Foo
+{
+    public function __construct($bar)
+    {
+        var_dump(2);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+namespace Baz {
+
+    class Foo
+    {
+        public function __construct($bar)
+        {
+            var_dump(1);
+        }
+
+        public function Foo($bar)
+        {
+            var_dump(2);
+        }
+    }
+}
+
+class Foo
+{
+    public function __construct($bar)
+    {
+        var_dump(2);
+    }
+}
+EOF;
+
+        $this->makeTest($input);
+    }
+
+    public function testPhp5Only()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($input);
+    }
+
+    public function testPhp4Only()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function Foo($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testBothRight()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    /**
+     * PHP-4 Constructor
+     */
+    function Foo($bar)
+    {
+        $this->__construct($bar);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testBothWrong()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * PHP-5 Constructor
+     */
+    function __construct($bar)
```

Comment:
See testAlphaBeta down below ;)

it would remove Foo with:
```php
$expected = <<<'EOF'
<?php
class Foo
{
    function __construct() { echo "alpha"; }
    function Foo() { $this->__construct(); echo "beta";  }
}
EOF;
```

But if you have something like this, then you deserve the code to be broken ;)